backend/src/services/tag.service.ts

```diff
@@ -31,19 +33,28 @@ export const getTags = async (): Promise<Tags[]> => {
 
 export const createTags = async (
   data: Tags['name'][],
-): Promise<MappedTag[]> => {
+): Promise<TagsCreationResponse> => {
   const tagsRepository = getCustomRepository(TagsRepository);
   const companyRepository = getCustomRepository(CompanyRepository);
 
   const companyInstance: Company = await companyRepository.findOne({
     name: companies[0].name,
   });
 
+  const targetTags = await tagsRepository.find({ company: companyInstance });
   const tags: Tags[] = [];
+  const existingTags: Tags['name'][] = [];
+
+  await asyncForEach(async (name) => {
+    const tagName = name.toLowerCase();
+
+    if (targetTags.find((tag) => tag.name === tagName)) {
+      existingTags.push(tagName);
+      return;
+    }
```

Comment:
You can cast to lower case during check, but save the original tag name to the database. If I created the tag "System Design", I want to see "System Design" in tags, not "system design"